state/factory/statedb.go

```diff
@@ -553,22 +556,20 @@ func (sdb *stateDB) createGenesisStates(ctx context.Context) error {
 
 // getFromWorkingSets returns (workingset, true) if it exists in a cache, otherwise generates new workingset and return (ws, false)
 func (sdb *stateDB) getFromWorkingSets(ctx context.Context, key hash.Hash256) (*workingSet, bool, error) {
-	sdb.mutex.RLock()
-	defer sdb.mutex.RUnlock()
 	if data, ok := sdb.workingsets.Get(key); ok {
 		if ws, ok := data.(*workingSet); ok {
 			// if it is already validated, return workingset
 			return ws, true, nil
 		}
 		return nil, false, errors.New("type assertion failed to be WorkingSet")
 	}
-	tx, err := sdb.newWorkingSet(ctx, sdb.currentChainHeight+1)
-
+	sdb.mutex.RLock()
+	h := sdb.currentChainHeight
+	sdb.mutex.RUnlock()
+	tx, err := sdb.newWorkingSet(ctx, h+1)
 	return tx, false, err
 }
 
 func (sdb *stateDB) putIntoWorkingSets(key hash.Hash256, ws *workingSet) {
-	sdb.mutex.Lock()
-	defer sdb.mutex.Unlock()
 	sdb.workingsets.Add(key, ws)
```

Comment:
This code is very straightforward. I think this func can be removed.

Reply:
fixed